### PR TITLE
Enable using -sign in VMR scenarios

### LINKED ISSUE #108605: Enable arcade -sign functionality in runtime
Right now runtime uses a bit of custom infra to sign (a separate job that gathers intermediate artifacts and signs them). For the VMR, it also needs to support the ability to do the traditional arcade-based signing. When -sign is passed, Sign.proj is invoked, items to sign is populated, etc. etc.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Enable signing using the Arcade signing infra in VMR scenarios. This does not move the dotnet-runtime-official pipeline to use the Arcade signing infra, similar to how https://github.com/dotnet/runtime/pull/99432 did not enable `-publish` in dotnet-runtime-official.

This PR adds the `-sign` flag to the inner invocation when the outer invocation is invoked with the `-sign` flag to ensure we go down the Arcade signing and publishing route as expected.

Based on the work in https://github.com/dotnet/runtime/pull/100004

Fixes https://github.com/dotnet/runtime/issues/108605

Runtime official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2561356&view=results

### CODE REVIEW

File: eng/DotNetBuild.props [hunk 1]
```diff
@@ -42,8 +42,9 @@
            This allows to build the repository using './build.sh <args> /p:DotNetBuildSourceOnly=true'.
            Properties that control flags from the VMR build, and the expected output for the VMR build should be added to this file. -->
 
-      <!-- Enable regular Arcade publishing in VMR build -->
+      <!-- Enable regular Arcade signing and publishing in VMR build -->
       <InnerBuildArgs Condition="'$(DotNetBuildOrchestrator)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)restore $(FlagParameterPrefix)build $(FlagParameterPrefix)publish</InnerBuildArgs>
+      <InnerBuildArgs Condition="'$(DotNetBuildOrchestrator)' == 'true' and '$(Sign)' == 'true'">$(InnerBuildArgs) $(FlagParameterPrefix)sign</InnerBuildArgs>
```
Comment (ViktorHofer) [inline eng/DotNetBuild.props:47]: nit: As the other actions above aren't conditioned, why condition on '$(Sign)' == 'true' here?

**Review by ViktorHofer — APPROVED** [review]
LGTM if it works

File: eng/DotNetBuild.props (hunk 1 above)
Comment (jkoritzinsky) [reply eng/DotNetBuild.props:47]: We don't want to pass down the `-sign` flag when the VMR isn't built with `-sign`. The rest of the flags are always passed down in the VMR (as we always restore/build/publish).

**Comment by jkoritzinsky** [comment]
Signing logs look good for the runtime official build. I'll merge this in so it can flow and catch up with https://github.com/dotnet/sdk/pull/44207, which enables some signing builds in the VMR.

**Comment by jkoritzinsky** [comment]
That may be. I'll look into it

**Comment by jkoritzinsky** [comment]
That error should be fixed by https://github.com/dotnet/runtime/pull/108990

**Pull request merged**